src/BMPDecoder.ts

```diff
@@ -0,0 +1,74 @@
+import { IOBuffer } from 'iobuffer';
+
+export default class BMPDecoder {
+  bufferData: IOBuffer;
+  pixelDataOffset: number;
+  width: number;
+  height: number;
+  bitDepth: number;
+
+  constructor(bufferData: Buffer) {
+    this.bufferData = new IOBuffer(bufferData);
+    const formatCheck = this.bufferData.readBytes(2);
+    if (formatCheck[0] !== 0x42 && formatCheck[1] !== 0x4d) {
+      throw new Error(
+        'This is not a BMP image or the encoding is not correct.'
+      );
+    }
+    this.pixelDataOffset = this.bufferData.skip(8).readUint32();
+    this.width = this.bufferData.skip(4).readUint32();
+    this.height = this.bufferData.readUint32();
+    this.bitDepth = this.bufferData.seek(28).readUint16();
```

Comment:
AFAIK we only support 1-bit depth, so it should throw here if it's not 1 and you should add a test with a an image that has a larger bit depth than 1.